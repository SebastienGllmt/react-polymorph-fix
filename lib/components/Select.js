'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createRef = require('create-react-ref/lib/createRef');

var _createRef2 = _interopRequireDefault(_createRef);

var _GlobalListeners = require('./HOC/GlobalListeners');

var _withTheme = require('./HOC/withTheme');

var _themes = require('../utils/themes');

var _ = require('.');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// internal components


// internal utility functions


// import constants


var SelectBase = function (_Component) {
  _inherits(SelectBase, _Component);

  // define static properties
  function SelectBase(props) {
    _classCallCheck(this, SelectBase);

    // define ref
    var _this = _possibleConstructorReturn(this, (SelectBase.__proto__ || Object.getPrototypeOf(SelectBase)).call(this, props));

    _this.focus = function () {
      return _this.toggleOpen();
    };

    _this.toggleOpen = function () {
      if (_this.state.isOpen && _this.optionsElement && _this.optionsElement.current) {
        // set Options scroll position to top on close
        _this.optionsElement.current.scrollTop = 0;
      }
      _this.setState({ isOpen: !_this.state.isOpen });
    };

    _this.toggleMouseLocation = function () {
      return _this.setState({ mouseIsOverOptions: !_this.state.mouseIsOverOptions });
    };

    _this.handleInputClick = function (event) {
      event.stopPropagation();
      event.preventDefault();

      var inputElement = _this.inputElement;

      if (inputElement.current && document.activeElement === inputElement.current) {
        inputElement.current.blur();
      }
      _this.toggleOpen();
    };

    _this.handleChange = function (option, event) {
      // check if the user passed an onChange handler and call it
      if (_this.props.onChange) _this.props.onChange(option.value, event);
      // onChange is called when an option is selected, so close options
      _this.toggleOpen();
    };

    _this.getSelectedOption = function () {
      var _this$props = _this.props,
          options = _this$props.options,
          value = _this$props.value,
          allowBlank = _this$props.allowBlank;
      var _iteratorNormalCompletion = true;
      var _didIteratorError = false;
      var _iteratorError = undefined;

      try {
        for (var _iterator = options[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
          var option = _step.value;

          if (option.value === value) return option;
        }
      } catch (err) {
        _didIteratorError = true;
        _iteratorError = err;
      } finally {
        try {
          if (!_iteratorNormalCompletion && _iterator.return) {
            _iterator.return();
          }
        } finally {
          if (_didIteratorError) {
            throw _iteratorError;
          }
        }
      }

      if (!allowBlank) return options[0];
    };

    _this.rootElement = (0, _createRef2.default)();
    _this.inputElement = (0, _createRef2.default)();
    _this.optionsElement = (0, _createRef2.default)();

    var context = props.context,
        themeId = props.themeId,
        theme = props.theme,
        themeOverrides = props.themeOverrides;


    _this.state = {
      composedTheme: (0, _themes.composeTheme)((0, _themes.addThemeId)(theme || context.theme, themeId), (0, _themes.addThemeId)(themeOverrides, themeId), context.ROOT_THEME_API),
      isOpen: false,
      mouseIsOverOptions: false
    };
    return _this;
  }
  // declare ref types


  _createClass(SelectBase, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      // check for autoFocus of input element
      if (this.props.autoFocus) {
        return this.focus();
      }
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      (0, _themes.didThemePropsChange)(this.props, nextProps, this.setState.bind(this));
    }

    // ========= PUBLIC SKIN API =========

    // applying focus to the input element will
    // toggle options open because Select's input is read only

  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      // destructuring props ensures only the "...rest" get passed down
      var _props = this.props,
          skin = _props.skin,
          theme = _props.theme,
          themeOverrides = _props.themeOverrides,
          autoFocus = _props.autoFocus,
          context = _props.context,
          allowBlank = _props.allowBlank,
          rest = _objectWithoutProperties(_props, ['skin', 'theme', 'themeOverrides', 'autoFocus', 'context', 'allowBlank']);

      var SelectSkin = skin || context.skins[_.IDENTIFIERS.SELECT];

      return _react2.default.createElement(
        _GlobalListeners.GlobalListeners,
        {
          mouseIsOverOptions: this.state.mouseIsOverOptions,
          optionsIsOpen: this.state.isOpen,
          optionsIsOpeningUpward: this.props.isOpeningUpward,
          optionsRef: this.optionsElement,
          rootRef: this.rootElement,
          toggleOpen: this.toggleOpen
        },
        function (_ref) {
          var optionsMaxHeight = _ref.optionsMaxHeight;
          return _react2.default.createElement(SelectSkin, _extends({
            isOpen: _this2.state.isOpen,
            rootRef: _this2.rootElement,
            inputRef: _this2.inputElement,
            optionsRef: _this2.optionsElement,
            optionsMaxHeight: optionsMaxHeight,
            theme: _this2.state.composedTheme,
            getSelectedOption: _this2.getSelectedOption,
            handleInputClick: _this2.handleInputClick,
            handleChange: _this2.handleChange,
            toggleOpen: _this2.toggleOpen,
            toggleMouseLocation: _this2.toggleMouseLocation
          }, rest));
        }
      );
    }
  }]);

  return SelectBase;
}(_react.Component);

SelectBase.displayName = 'Select';
SelectBase.defaultProps = {
  allowBlank: true,
  autoFocus: false,
  context: (0, _withTheme.createEmptyContext)(),
  isOpeningUpward: false,
  options: [],
  theme: null,
  themeOverrides: {},
  themeId: _.IDENTIFIERS.SELECT,
  value: ''
};
var Select = exports.Select = (0, _withTheme.withTheme)(SelectBase);
//# sourceMappingURL=Select.js.map