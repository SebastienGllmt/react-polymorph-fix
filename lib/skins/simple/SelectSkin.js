'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectSkin = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Options = require('../../components/Options');

var _Input = require('../../components/Input');

var _OptionsSkin = require('./OptionsSkin');

var _InputSkin = require('./InputSkin');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// skins


// components
var SelectSkin = exports.SelectSkin = function SelectSkin(props) {
  var selectedOption = props.getSelectedOption();
  var inputValue = selectedOption ? selectedOption.label : '';
  var theme = props.theme,
      themeId = props.themeId;

  return _react2.default.createElement(
    'div',
    {
      ref: props.rootRef,
      className: (0, _classnames2.default)([props.className, theme[themeId].select, props.isOpen ? theme[themeId].isOpen : null, props.isOpeningUpward ? theme[themeId].openUpward : null])
    },
    _react2.default.createElement(
      'div',
      { className: theme[themeId].selectInput },
      _react2.default.createElement(_Input.Input, {
        skin: _InputSkin.InputSkin,
        theme: theme,
        inputRef: props.inputRef,
        label: props.label,
        value: inputValue,
        onClick: props.handleInputClick,
        placeholder: props.placeholder,
        error: props.error,
        readOnly: true
      })
    ),
    _react2.default.createElement(_Options.Options, {
      skin: _OptionsSkin.OptionsSkin,
      theme: theme,
      isOpen: props.isOpen,
      optionsRef: props.optionsRef,
      optionsMaxHeight: props.optionsMaxHeight,
      options: props.options,
      isOpeningUpward: props.isOpeningUpward,
      onChange: props.handleChange,
      optionRenderer: props.optionRenderer,
      selectedOption: selectedOption,
      noResults: !props.options.length,
      targetRef: props.inputRef,
      toggleMouseLocation: props.toggleMouseLocation,
      toggleOpen: props.toggleOpen
    })
  );
};

// external libraries
//# sourceMappingURL=SelectSkin.js.map