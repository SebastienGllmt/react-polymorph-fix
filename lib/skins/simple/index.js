'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map