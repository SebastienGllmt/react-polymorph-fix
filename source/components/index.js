// @flow
export const IDENTIFIERS = {
  AUTOCOMPLETE: 'autocomplete',
  BUBBLE: 'bubble',
  BUTTON: 'button',
  CHECKBOX: 'checkbox',
  FLEX: 'flex',
  FORM_FIELD: 'formfield',
  GRID: 'grid',
  GUTTER: 'gutter',
  HEADER: 'header',
  INFINITE_SCROLL: 'infinitescroll',
  INPUT: 'input',
  LOADING_SPINNER: 'loadingspinner',
  MODAL: 'modal',
  OPTIONS: 'options',
  PROGRESS_BAR: 'progressbar',
  RADIO: 'radio',
  SELECT: 'select',
  SWITCH: 'switch',
  TEXT_AREA: 'textarea',
  TOGGLER: 'toggler',
  TOOLTIP: 'tooltip'
};
