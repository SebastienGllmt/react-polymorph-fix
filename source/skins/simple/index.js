// @flow
import { IDENTIFIERS } from '../../components';
import { AutocompleteSkin } from './AutocompleteSkin';
import { BubbleSkin } from './BubbleSkin';
import { ButtonSkin } from './ButtonSkin';
import { CheckboxSkin } from './CheckboxSkin';
import { FormFieldSkin } from './FormFieldSkin';
import { HeaderSkin } from './HeaderSkin';
import { InfiniteScrollSkin } from './InfiniteScrollSkin';
import { InputSkin } from './InputSkin';
import { LoadingSpinnerSkin } from './LoadingSpinnerSkin';
import { ModalSkin } from './ModalSkin';
import { OptionsSkin } from './OptionsSkin';
import { ProgressBarSkin } from './ProgressBarSkin';
import { RadioSkin } from './RadioSkin';
import { SelectSkin } from './SelectSkin';
import { SwitchSkin } from './SwitchSkin';
import { TextAreaSkin } from './TextAreaSkin';
import { TogglerSkin } from './TogglerSkin';
import { TooltipSkin } from './TooltipSkin';

export const SimpleSkins = {
  [IDENTIFIERS.AUTOCOMPLETE]: AutocompleteSkin,
  [IDENTIFIERS.BUBBLE]: BubbleSkin,
  [IDENTIFIERS.BUTTON]: ButtonSkin,
  [IDENTIFIERS.CHECKBOX]: CheckboxSkin,
  [IDENTIFIERS.FORM_FIELD]: FormFieldSkin,
  [IDENTIFIERS.HEADER]: HeaderSkin,
  [IDENTIFIERS.INFINITE_SCROLL]: InfiniteScrollSkin,
  [IDENTIFIERS.INPUT]: InputSkin,
  [IDENTIFIERS.LOADING_SPINNER]: LoadingSpinnerSkin,
  [IDENTIFIERS.MODAL]: ModalSkin,
  [IDENTIFIERS.OPTIONS]: OptionsSkin,
  [IDENTIFIERS.PROGRESS_BAR]: ProgressBarSkin,
  [IDENTIFIERS.RADIO]: RadioSkin,
  [IDENTIFIERS.SELECT]: SelectSkin,
  [IDENTIFIERS.SWITCH]: SwitchSkin,
  [IDENTIFIERS.TEXT_AREA]: TextAreaSkin,
  [IDENTIFIERS.TOGGLER]: TogglerSkin,
  [IDENTIFIERS.TOOLTIP]: TooltipSkin
};
