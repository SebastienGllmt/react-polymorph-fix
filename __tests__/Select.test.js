import React from 'react';
import renderer from 'react-test-renderer';

import { Select } from '../source/components/Select';
import { SelectSkin } from '../source/skins/simple/SelectSkin';
import { renderInSimpleTheme } from './helpers/theming';

const COUNTRIES = [
  { label: 'Frankreich', value: 'France' },
  { label: 'Spanien', value: 'Spain' },
  { label: 'Kroatien', value: 'Croatia' },
  { label: 'Vereinigten Staaten', value: 'United States' },
  { label: 'Österreich', value: 'Austria' }
];

const COUNTRIES_DISABLED_OPTIONS = [
  { value: 'EN-gb', label: 'England' },
  { value: 'ES-es', label: 'Spain', isDisabled: true },
  { value: 'TH-th', label: 'Thailand', isDisabled: true },
  { value: 'EN-en', label: 'USA' }
];

test('Select renders correctly', () => {
  const component = renderInSimpleTheme(
    <Select
      options={COUNTRIES}
      skin={SelectSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select renders with placeholder', () => {
  const component = renderInSimpleTheme(
    <Select
      placeholder="Select your country …"
      options={COUNTRIES}
      skin={SelectSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select renders with an error', () => {
  const component = renderInSimpleTheme(
    <Select
      error="Please select a different option"
      options={COUNTRIES}
      skin={SelectSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select renders with disabled options', () => {
  const component = renderInSimpleTheme(
    <Select
      options={COUNTRIES_DISABLED_OPTIONS}
      skin={SelectSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select isOpeningUpward={true}', () => {
  const component = renderInSimpleTheme(
    <Select
      isOpeningUpward
      options={COUNTRIES}
      skin={SelectSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select isOpen={true}', () => {
  const component = renderInSimpleTheme(
    <Select
      isOpen
      options={COUNTRIES}
      skin={SelectSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select uses render prop - optionRenderer', () => {
  const component = renderInSimpleTheme(
    <Select
      options={COUNTRIES}
      skin={SelectSkin}
      optionRenderer={option => (
        <div>
          <span>German: {option.label}</span>
          <span>English: {option.value}</span>
        </div>
      )}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
