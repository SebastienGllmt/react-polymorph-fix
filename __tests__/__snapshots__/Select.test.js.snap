// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select isOpen={true} 1`] = `
<div>
  <div
    className="select isOpen"
  >
    <div
      className="selectInput"
    >
      <div
        className="root"
      >
        
        <div
          className="inputWrapper"
        >
          <input
            className="input"
            onChange={[Function]}
            onClick={[Function]}
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options isOpen firstOptionHighlighted root isFloating"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Frankreich
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Spanien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Kroatien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Vereinigten Staaten
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Österreich
            </span>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;

exports[`Select isOpeningUpward={true} 1`] = `
<div>
  <div
    className="select openUpward"
  >
    <div
      className="selectInput"
    >
      <div
        className="root"
      >
        
        <div
          className="inputWrapper"
        >
          <input
            className="input"
            onChange={[Function]}
            onClick={[Function]}
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options openUpward root openUpward isFloating isHidden"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Österreich
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Vereinigten Staaten
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Kroatien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Spanien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Frankreich
            </span>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 1`] = `
<div>
  <div
    className="select"
  >
    <div
      className="selectInput"
    >
      <div
        className="root"
      >
        
        <div
          className="inputWrapper"
        >
          <input
            className="input"
            onChange={[Function]}
            onClick={[Function]}
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options firstOptionHighlighted root isFloating isHidden"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Frankreich
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Spanien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Kroatien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Vereinigten Staaten
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Österreich
            </span>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;

exports[`Select renders with an error 1`] = `
<div>
  <div
    className="select"
  >
    <div
      className="selectInput"
    >
      <div
        className="root errored"
      >
        <div
          className="error"
        >
          Please select a different option
        </div>
        <div
          className="inputWrapper"
        >
          <input
            className="input errored"
            onChange={[Function]}
            onClick={[Function]}
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options firstOptionHighlighted root isFloating isHidden"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Frankreich
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Spanien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Kroatien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Vereinigten Staaten
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Österreich
            </span>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;

exports[`Select renders with disabled options 1`] = `
<div>
  <div
    className="select"
  >
    <div
      className="selectInput"
    >
      <div
        className="root"
      >
        
        <div
          className="inputWrapper"
        >
          <input
            className="input"
            onChange={[Function]}
            onClick={[Function]}
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options firstOptionHighlighted root isFloating isHidden"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              England
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option disabledOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Spain
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option disabledOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Thailand
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              USA
            </span>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;

exports[`Select renders with placeholder 1`] = `
<div>
  <div
    className="select"
  >
    <div
      className="selectInput"
    >
      <div
        className="root"
      >
        
        <div
          className="inputWrapper"
        >
          <input
            className="input"
            onChange={[Function]}
            onClick={[Function]}
            placeholder="Select your country …"
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options firstOptionHighlighted root isFloating isHidden"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Frankreich
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Spanien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Kroatien
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Vereinigten Staaten
            </span>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <span
              className="label"
            >
              Österreich
            </span>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;

exports[`Select uses render prop - optionRenderer 1`] = `
<div>
  <div
    className="select"
  >
    <div
      className="selectInput"
    >
      <div
        className="root"
      >
        
        <div
          className="inputWrapper"
        >
          <input
            className="input"
            onChange={[Function]}
            onClick={[Function]}
            readOnly={true}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="options firstOptionHighlighted root isFloating isHidden"
      style={null}
    >
      <div
        className="bubble"
        data-bubble-container={true}
      >
        <ul
          className="ul"
        >
          <li
            aria-hidden={true}
            className="option highlightedOption"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <div>
              <span>
                German: 
                Frankreich
              </span>
              <span>
                English: 
                France
              </span>
            </div>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <div>
              <span>
                German: 
                Spanien
              </span>
              <span>
                English: 
                Spain
              </span>
            </div>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <div>
              <span>
                German: 
                Kroatien
              </span>
              <span>
                English: 
                Croatia
              </span>
            </div>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <div>
              <span>
                German: 
                Vereinigten Staaten
              </span>
              <span>
                English: 
                United States
              </span>
            </div>
          </li>
          <li
            aria-hidden={true}
            className="option"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="presentation"
          >
            <div>
              <span>
                German: 
                Österreich
              </span>
              <span>
                English: 
                Austria
              </span>
            </div>
          </li>
        </ul>
      </div>
      <span
        className="arrow"
        data-bubble-arrow={true}
      />
    </div>
  </div>
</div>
`;
